ONSdigital/ssdc-rm-acceptance-tests
# SDCSRM-512 Adding script to check support tool is available

# Motivation and Context

We've seen a lot of times recently that the ATs fail to start due to the Support Tool ingress not being available. We want a script that will poll the Support Tool health endpoint to check that the ingress is available for the Acceptance Tests to start using.
# What has changed



- Added script to poll support tool health endpoint
# How to test?



- You can test locally by stopping the support tool docker container and run the command below. This will poll the health endpoint 10 times which should last 10 minutes. While the script is running, start the container and you should see the script finish since it can hit the endpoint
```
pipenv run python -m acceptance_tests.utilities.support_tool_healthcheck --url http://localhost:9999/actuator/health --max_retries 10
```
- Run this in your GCP project. I've added this to the run_gke script which just passes in the $ENV variable into what the support tool url should be. This should check that support tool is up and then start running the tests when it can hit the endpoint
# Links



[Jira](https://jira.ons.gov.uk/browse/SDCSRM-512)
